langchain/agents/conversational/base.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+import re
+from typing import Any, Callable, List, NamedTuple, Optional, Tuple
+
+from langchain.agents.agent import Agent
+from langchain.agents.mrkl.prompt import FORMAT_INSTRUCTIONS
+from langchain.agents.conversational.prompt import PREFIX, SUFFIX
+from langchain.agents.tools import Tool
+from langchain.llms.base import BaseLLM
+from langchain.prompts import PromptTemplate
+
+
+class ConversationalAgent(Agent):
+    """Agent for the MRKL chain."""
+
+    @property
+    def observation_prefix(self) -> str:
+        """Prefix to append the observation with."""
+        return "Observation: "
+
+    @property
+    def llm_prefix(self) -> str:
+        """Prefix to append the llm call with."""
+        return "Thought:"
+
+    @classmethod
+    def create_prompt(
+        cls,
+        tools: List[Tool],
+        prefix: str = PREFIX,
+        suffix: str = SUFFIX,
+        input_variables: Optional[List[str]] = None,
+    ) -> PromptTemplate:
+        """Create prompt in the style of the zero shot agent.
+
+        Args:
+            tools: List of tools the agent will have access to, used to format the
+                prompt.
+            prefix: String to put before the list of tools.
+            suffix: String to put after the list of tools.
+            input_variables: List of input variables the final prompt will expect.
+
+        Returns:
+            A PromptTemplate with the template assembled from the pieces here.
+        """
+        tool_strings = "\n".join([f"{tool.name}: {tool.description}" for tool in tools])
+        tool_names = ", ".join([tool.name for tool in tools])
+        format_instructions = FORMAT_INSTRUCTIONS.format(tool_names=tool_names)
```

Comment:
This seems to result in the final answer when a tool is used to come formatted as `Final Answer: ARR yesterday was 3500.0.`. Which then fails in `_extract_tool_and_input` as it expects something like `AI: ARR yesterday was 3500.0.` From my testing it can be fixed by customising `FORMAT_INSTRUCTIONS` to end in `AI: the final answer to the original input question`